### PR TITLE
Integrate tauri-upload-plugin

### LINKED ISSUE #3: Add Rust API for file download with progress
We need a Rust API that lets us download large (4GB+) model files efficiently. There's a chance we will want to use this for more than just models so it should be a general purpose API.

- [ ] Download large files efficiently using streaming
- [ ] Report progress back to the UI
- [ ] Allow the UI to specify a download location from [fs BaseDirectory enum](https://tauri.app/v1/api/js/fs).

Relevant discussion: https://github.com/tauri-apps/tauri/discussions/4726

### PR DESCRIPTION
**Pull request by ospfranco**
Adds an API callable from JS that allows to download/upload files

closes #3

### CODE REVIEW

File: src/main.tsx [hunk 1]
```diff
@@ -3,6 +3,8 @@ import App from './app/App';
 import '@/shared/styles/index.scss';
 import Llamafile from './llamafile';
 import { ThemeProvider, ErrorBoundary } from '@/app/providers';
+import { database } from '../db.tsx';
```
Comment (ospfranco) [inline src/main.tsx:6]: I think this changed slipped. Let me know so I can revert it.

File: src-tauri/Cargo.toml [hunk 2]
```diff
@@ -11,9 +11,10 @@ edition = "2021"
 tauri-build = { version = "1.5", features = [] }
 
 [dependencies]
-tauri = { version = "1.5", features = [ "shell-sidecar"] }
```
Comment (ospfranco) [inline src-tauri/Cargo.toml:14]: I guess only I will be doing Rust for the time being. So this is just automatic linting done by VSCode and the rust linter.

File: src/main.tsx (hunk 1 above)
Comment (demetrxx) [reply src/main.tsx:6]: Yeah, I believe this one is extra

**Review by demetrxx — CHANGES_REQUESTED** [review]
LGTM! 
Just 1 minor change

**Review by nchapman — APPROVED** [review]
LGTM 👍

**Comment by ospfranco** [comment]
Merged

**Pull request merged**